### DISCUSSION TITLE
How to generate new addresses for wallet in Shelley protocol?

### DISCUSSION THREAD

**Discussion by yurayakimenko** [cardano-foundation/cardano-wallet#4044] [Q&A] [answered]
Hi! Please, could anyone explain, how to generate new address for Cardano wallet on Shelley version of protocol? I'm using official docker image of cardano-wallet and it's API docs. But it's unclear to me, where to get possible credentials with prefixes: 'script_vkh', 'script_vk' or 'script_xvk' ? 

Shelley Wallet API docs got paragraph (https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/postAnyAddress), which is called Construct Address. I don't understand, why I can't generate address in a simple way as it was in Byron version of the protocol which required only walletId, passphrase and address_index?

There are three arguments to generate address - payment, stake and validation, but nobody explained in docs how to generate related public keys. I've tried method `POST /wallets/{walletId}/keys/{index}` but obtained key was started with acct_xvk...

And if I try to pass this key to generate address requests I got this error:

`Error in $: Address must have at least one valid credential. When script is used as a credential it must have only bech32 encoded verification keys with possible prefixes: 'script_vkh', 'script_vk' or 'script_xvk' and proper payload size. 'at_least'cannot exceed 255. When public key is used as a credential then bech32 encoded public keys are expected to be used with possible prefixes: 'stake_vk' or 'addr_vk', always with proper payload size.`

Despite this, you closed support to previous versions of protocol and attempts to run CLI command also failed:

```
$ docker exec -it <container name> cardano-wallet address create <walletId>
$ cardano-wallet: user error (feature unavailable.)
```

I'm writing here because I've been ignored in official telegram group and I understand that GitHub issues is not StackOverflow, but it seems like no one else except original developers can help me.

P.S. I understand that wallet generates new address after each receive transaction but I need to generate them manually.

Thanks in advance!

**Comment by KtorZ** [comment]
Hello, the short answer is that: you can't. 

The long answer is written with more details in the following Cardano Improvement Proposal https://github.com/cardano-foundation/CIPs/blob/master/CIP-1852/CIP-1852.md.

**Comment by tnsantana** [comment]
Is it possible to import keys or addresses generated with cardano-addresses? Any solution for utxos and balance management with enterprise addresses?

**Comment by HeinrichApfelmus** [comment, marked as answer]
@tnsantana Unfortunately not, as this affects the ability to restore the wallet from a single seed mnemonic (payments to or from imported addresses would be missed when restoring). This is a technical limitation of industry standard BIP-44 on which Cardano CIP-1852 is based.
